### PR TITLE
build(deps): consolidate Dependabot updates

### PR DESCRIPTION
**Pull request by NormannK**
## Summary

Consolidates eleven Dependabot updates into one reviewable change:

### Runtime dependencies

- `cachebox` 6.2.5 → 6.2.6 (#1267)
- `linkify-it-py` 2.1.0 → 2.2.0 (#1272)
- `pydantic` 2.13.4 → 2.13.5 (#1268)
- `platformdirs` 4.11.3 → 4.11.5 (#1270)
- `python-fasthtml` 0.14.11 → 0.14.12 (#1259)
- `statsmodels` 0.14.6 → 0.15.0 (#1263)

### Development and CI dependencies

- `commitizen` 4.17.1 → 4.18.0 (#1260)
- `GitPython` 3.1.59 → 3.1.61 (#1269)
- `types-docutils` 0.22.3.20260712 → 0.23.0.20260827 (#1264)
- `coverage` 7.15.4 → 7.16.0 (#1265)
- `pypdf` 6.16.1 → 6.16.2 (#1271)

The `uv.lock` file is regenerated for all updates. The isolated mypy dependency in `.pre-commit-config.yaml` is also synchronized to `types-docutils==0.23.0.20260827`.

### Runtime impact

- **cachebox:** low-risk patch update. Improves TTL/VTTL background sweeping, fixes asynchronous cache hit/miss tracking, modernizes typing, and adds Windows ARM64 wheels. Cache timing and metrics paths deserve smoke coverage.
- **linkify-it-py:** security-relevant minor update. It includes the 2.1.1 fix for quadratic processing of untrusted linkified input, improves astral-plane Unicode link handling, changes some email-linkification behavior, and removes the `uc-micro-py` runtime dependency. Markdown/linkification tests should cover long hostile input, Unicode hostnames, and email edge cases.
- **pydantic:** low-to-moderate-risk patch update. Fixes validator reuse, garbage-collector traversal, and smart-union field counting. This can correct model-selection behavior for ambiguous unions; API and configuration-model tests are relevant.
- **platformdirs:** low-risk patch update affecting platform-specific filesystem path resolution. Cache/config/data-directory tests are the relevant validation area.
- **python-fasthtml:** low-risk patch update fixing route argument handling and signature annotations; expected to improve web UI behavior.
- **statsmodels:** the primary compatibility risk. Version 0.15.0 standardizes randomness parameters around `rng`, changes some tuple results to named tuples, tightens option validation, raises dependency floors, and can change numerical output in corrected model paths. EOS's pinned NumPy, SciPy, and pandas versions satisfy the declared floors, but forecasting and optimization regression tests remain important.

### Development and CI impact

- **commitizen:** adds commit body-length limiting support; no production runtime impact.
- **GitPython:** patch update used for documentation/development workflows. Includes repository-discovery fixes and regex compatibility changes; no production runtime impact in the current dependency grouping.
- **types-docutils:** type-stub update, now synchronized with the pre-commit mypy hook. It may expose new type-check findings but has no production runtime impact.
- **coverage:** improves cross-platform coverage-data combining and configuration validation; no production runtime impact, though CI coverage diagnostics may change.
- **pypdf:** patch update used by tests/development; no production runtime impact in the current dependency grouping.

## Transitive lockfile impact

Regenerating `uv.lock` adds `formulaic` and `interface-meta` for statsmodels 0.15.0, updates `pydantic-core` for pydantic 2.13.5, and removes `uc-micro-py` after the linkify-it-py update.

## Risk and validation

Overall risk is **moderate**, driven primarily by statsmodels 0.15.0. The linkify-it-py update reduces a known denial-of-service risk but has small linkification behavior changes. Validate:

- full unit and integration test suite;
- mypy and all pre-commit hooks;
- forecasting and optimization numerical regression tests;
- web/API, cache, Markdown, and linkification smoke tests;
- locked/frozen environment installation;
- coverage collection across supported environments.

Closes #1259.
Closes #1260.
Closes #1263.
Closes #1264.
Closes #1265.
Closes #1267.
Closes #1268.
Closes #1269.
Closes #1270.
Closes #1271.
Closes #1272.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -24,7 +24,7 @@ dependencies = [
     "fastapi[standard-no-fastapi-cloud-cli]==0.141.1",
     "fastapi_cli==0.0.32",
     "rich-toolkit==0.20.3",
-    "python-fasthtml==0.14.11",
+    "python-fasthtml==0.14.12",
```
[the remark this answers is not in the document]
Comment (NormannK) [reply pyproject.toml:27]: Fixed in fa0c8f1. Regenerated and committed `uv.lock` for all ten rolled-up dependency updates; `uv lock --check` succeeds. The lock now includes the requested versions and required transitive updates.

File: pyproject.toml [hunk 2]
```diff
@@ -64,9 +64,9 @@ dev = [
     "types-requests==2.33.0.20260712", # for mypy
     "pandas-stubs==3.0.5.260730", # for mypy
     "tokenize-rt==6.2.0", # for mypy
-    "types-docutils==0.22.3.20260712", # for mypy
+    "types-docutils==0.23.0.20260827", # for mypy
```
[the remark this answers is not in the document]
Comment (NormannK) [reply pyproject.toml:67]: Fixed in fa0c8f1. Updated `.pre-commit-config.yaml` so the isolated mypy hook now uses `types-docutils==0.23.0.20260827`, matching `pyproject.toml` and `uv.lock`.

**Comment by NormannK** [comment]
assigned as a FYI.
